File: ParadigmaComp/15Questoes/python/questao8.py
# Agregação

class Empregado:
    def __init__(self, nome, cargo, salario):
        self.nome = nome
        self.cargo = cargo
        self.salario = salario

class Empresa:
    def __init__(self, nome):
        self.nome = nome
        self.empregados = []

    def adicionar_empregado(self, empregado):
        self.empregados.append(empregado)

    def exibir_empregados(self):
        for empregado in self.empregados:
            print(f"Nome: {empregado.nome}, Cargo: {empregado.cargo}, Salário: {empregado.salario}")

# Exemplo de uso
empresa = Empresa("Tech Corp")
empregado1 = Empregado("Alice", "Desenvolvedora", 5000)
empregado2 = Empregado("Bob", "Gerente", 7000)

empresa.adicionar_empregado(empregado1)
empresa.adicionar_empregado(empregado2)

empresa.exibir_empregados()

File: ParadigmaComp/15Questoes/golang/questao8.go
package main

// Agregação

import "fmt"

// Estrutura Empregado
type Empregado struct {
	Nome   string
	Cargo  string
	Salario float64
}

// Estrutura Empresa
type Empresa struct {
	Nome      string
	Empregados []Empregado
}

func (e *Empresa) AdicionarEmpregado(empregado Empregado) {
	e.Empregados = append(e.Empregados, empregado)
}

func (e Empresa) ExibirEmpregados() {
	for _, empregado := range e.Empregados {
		fmt.Printf("Nome: %s, Cargo: %s, Salário: %.2f\n", empregado.Nome, empregado.Cargo, empregado.Salario)
	}
}

func main() {
	empresa := Empresa{Nome: "Tech Corp"}
	empregado1 := Empregado{"Alice", "Desenvolvedora", 5000}
	empregado2 := Empregado{"Bob", "Gerente", 7000}

	empresa.AdicionarEmpregado(empregado1)
	empresa.AdicionarEmpregado(empregado2)

	empresa.ExibirEmpregados()
}
